packages/web/src/shared/components/form/RichTextInput.tsx

```diff
@@ -0,0 +1,179 @@
+import { useEditor, EditorContent } from '@tiptap/react';
+import StarterKit from '@tiptap/starter-kit';
+import { Icon } from '../Icon';
+import { ComponentProps, forwardRef, useImperativeHandle, useRef } from 'react';
+
+export interface RichTextInputProps {
+  name: string;
+  onChange?(e: { target: HTMLInputElement }): void;
+  onBlur?(e: { target: HTMLInputElement }): void;
+  'aria-labelledby'?: string;
+  'aria-label'?: string;
+}
+
+export interface RichTextInputRef {
+  focus(): void;
+}
+
+export const extensions = [
+  StarterKit.configure({
+    code: false,
+    codeBlock: false,
+    blockquote: false,
+    heading: false,
+    horizontalRule: false,
+  }),
+];
+
+const RichTextInput = forwardRef<RichTextInputRef, RichTextInputProps>(
+  ({ name, onChange, onBlur, ...props }, ref) => {
+    const hiddenInput = useRef<HTMLInputElement>(null);
+
+    const editor = useEditor({
+      extensions,
+      editorProps: {
+        attributes: {
+          class: 'focus:outline-none min-h-[24px] rich-text',
+          ...props,
+        },
+      },
+      onCreate({ editor }) {
+        const input = hiddenInput.current;
+        if (input) {
+          input.value = editor.getHTML();
+        }
+      },
+      onUpdate({ editor }) {
+        const input = hiddenInput.current;
+        if (input) {
+          input.value = editor.getHTML();
+          onChange?.({ target: input });
+        }
+      },
+      onBlur() {
+        const input = hiddenInput.current;
+        if (input) {
+          onBlur?.({ target: input });
+        }
+      },
+    });
+
+    // We expose a ref that can be consumed by react hook form without a Controller.
+    // This requires the ability to set the value, and focus.
+    useImperativeHandle(ref, () => ({
+      get value() {
+        return editor?.getHTML();
+      },
+      set value(value: string | undefined) {
+        if (editor?.getHTML() === value) return;
+
+        editor?.commands.setContent(value ?? '', false);
+        const input = hiddenInput.current;
+        if (input) {
+          input.value = value ?? '';
+        }
+      },
+      focus() {
+        editor?.commands.focus();
+      },
+    }));
+
+    return (
+      <div className="border rounded border-slate-400 focus-within:outline focus-within:outline-2 focus-within:outline-blue-600">
+        <input type="hidden" ref={hiddenInput} name={name} />
```

Comment:
This hidden input means that the RichTextInput can be used uncontrolled. This means we aren't storing the input value in react state (ie useState). So this component syncs the rich text html to this hidden input. This plus the use of `useImperativeHandle` to provide a custom ref means that react-hook-form can treat this component like a standard `input` element rather than a controlled component which would require `<Controller />`